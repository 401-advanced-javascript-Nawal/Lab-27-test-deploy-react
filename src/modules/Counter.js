import React from 'react';

class Counter extends React.Component {
  constructor(props) {
    super(props);
    // initial counter
    this.state = {
      counter: 0,
    };
  }
  
  
    handleClickUp = e => {
      e.preventDefault();
      let counter = this.state.counter;
      counter++;
      this.setState({ counter });
    }
  
    handleClickDown = e => {
      e.preventDefault();
      let counter = this.state.counter;
      counter--;
      this.setState({ counter });
    }
  
    render() {
      return (
        <section>
          <h3>{this.state.counter}</h3>
          <button id="b1" onClick={this.handleClickUp}>Click For Up</button>
          <button id="b2" onClick={this.handleClickDown}>Click For Down</button>
        </section>
      );
    }
}

export default Counter;