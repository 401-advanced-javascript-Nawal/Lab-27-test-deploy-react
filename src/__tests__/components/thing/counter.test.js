import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../../../modules/Counter.js';


describe ('<Counter />' , () => {

    it(' It Should be exists in our applications ', () => {
        // virtual 
        let app = shallow(<Counter />);
        // find the tag which already includes in our app 
        expect(app.find('h3').exists()).toBeTruthy();
    }) // end of it test 

    it(' It Should be change the state on click for Button 1 to increase the counter by 1 ', () => {
        // virtual 
        let app = mount(<Counter />);
        let button = app.find('#b1');
        button.simulate('click');
        expect(app.state('counter')).toEqual(1);
    }) // end of it button1

    it(' It Should be change the state on click for Button 1 to decrease the counter by 1 ', () => {
        // virtual 
        let app = mount(<Counter />);
        let button = app.find('#b2');
        button.simulate('click');
        expect(app.state('counter')).toEqual(-1);
    }) // end of it for button2

    it(' Renders correctly ' , () => {
        const tree = renderer.create(<Counter />).toJSON();
        expect(tree).toMatchSnapshot();
    }) // end of it test for render 

}); // end of describe test 