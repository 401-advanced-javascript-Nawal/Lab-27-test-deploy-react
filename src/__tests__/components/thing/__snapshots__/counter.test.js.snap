// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter />  Renders correctly  1`] = `
<section>
  <h3>
    0
  </h3>
  <button
    id="b1"
    onClick={[Function]}
  >
    Click For Up
  </button>
  <button
    id="b2"
    onClick={[Function]}
  >
    Click For Down
  </button>
</section>
`;
